base_rest/models/rest_service_registration.py

```diff
@@ -48,6 +57,52 @@ def _register_hook(self):
             (name, cls) for name, cls in controllers if "RestController" not in name
         ]
         http.controllers_per_module["base_rest"] = controllers
+        # create the final controller providing the http routes for
+        # the services available into the current database
+        self._build_controllers_routes(services_registry)
+
+    def _build_controllers_routes(self, services_registry):
+        for controller_def in services_registry.values():
+            for service in self._get_services(controller_def["collection_name"]):
+                RestApiMethodTransformer(service, controller_def).fix()
+                self._build_controller(service, controller_def)
+
+    def _build_controller(self, service, controller_def):
+        base_controller_cls = controller_def["controller_class"]
+        # build our new controller class
+        ctrl_cls = RestApiServiceControllerGenerator(
+            service, base_controller_cls
+        ).generate()
+
+        # generate an addon name used to register our new controller for
+        # the current database
+        addon_name = "{}_{}".format(
```

Comment:
shouldn't belong to the service's original module?

Reply:
Since services can be inherited and to avoid having to search in which addon the service was first declared, it is easier to build this name on the basis of the service's information. The goal is to have 1 controller per service. Reading the code again, it makes me say that the name of the collection should also be included in the name of the addon because the same service can be declared in several collections....